tests/Resources/ti.ui.tableview.test.js

```diff
@@ -0,0 +1,1439 @@
+				tableView.insertRowAfter(0, { title: 'White' });
+
+				finish();
+			} catch (err) {
+				return finish(err);
+			}
+		});
+
+		win.add(tableView);
+		win.open();
+	});
+
+	it.android('SearchView persistence', function (finish) {
+		var	tableData = [ { title: 'Apples' }, { title: 'Bananas' }, { title: 'Carrots' }, { title: 'Potatoes' } ],
+			searchView = Ti.UI.Android.createSearchView(),
+			table = Ti.UI.createTableView({
+				height: '80%',
+				search: searchView,
+				data: tableData
+			});
+		win = Ti.UI.createWindow();
+		function removeAndAddTable() {
+			try {
+				table.removeEventListener('postlayout', removeAndAddTable);
+				win.remove(table);
+				win.add(table);
+				finish();
+			} catch (err) {
+				finish(err);
+			}
+		}
+
+		table.addEventListener('postlayout', removeAndAddTable);
+		win.add(table);
+		win.open();
+	});
+
+	it('row#color row#backgroundColor', function (finish) {
+		// Set up a TableView with colored rows.
+		win = Ti.UI.createWindow();
+		const section = Ti.UI.createTableViewSection({ headerTitle: 'Section' });
+		const row1 = Ti.UI.createTableViewRow({
+			title: 'Row 1',
+			color: 'white',
+			backgroundColor: 'blue'
+		});
+		const row2 = Ti.UI.createTableViewRow({
+			title: 'Row 2',
+			color: 'black',
+			backgroundColor: 'yellow'
+		});
+		section.add(row1);
+		section.add(row2);
+		const tableView = Ti.UI.createTableView({ data: [ section ] });
+		win.add(tableView);
+
+		// Verify row objects return same color values assigned above.
+		should(row1.color).be.eql('white');
+		should(row1.backgroundColor).be.eql('blue');
+		should(row2.color).be.eql('black');
+		should(row2.backgroundColor).be.eql('yellow');
+
+		// Open window to test dynamic color changes.
+		win.addEventListener('open', function () {
+			row1.color = 'red';
+			row1.backgroundColor = 'white';
+			row2.color = 'white';
+			row2.backgroundColor = 'purple';
+			setTimeout(function () {
+				should(row1.color).be.eql('red');
+				should(row1.backgroundColor).be.eql('white');
+				should(row2.color).be.eql('white');
+				should(row2.backgroundColor).be.eql('purple');
+				finish();
+			}, 1);
+		});
+		win.open();
+	});
+
+	it('row - read unassigned color properties', function (finish) {
+		win = Ti.UI.createWindow();
+		const section = Ti.UI.createTableViewSection({ headerTitle: 'Section' });
+		const row1 = Ti.UI.createTableViewRow({ title: 'Row 1' });
+		section.add(row1);
+		const tableView = Ti.UI.createTableView({ data: [ section ] });
+		win.add(tableView);
+
+		function validateRow() {
+			// Verify we can read row color properties without crashing. (Used to crash on Android.)
+			// We don't care about the returned value.
+			// eslint-disable-next-line no-unused-vars
+			let value;
+			value = row1.color;
+			value = row1.backgroundColor;
+			if (Ti.Android) {
+				value = row1.backgroundDisabledColor;
+				value = row1.backgroundFocusedColor;
+				value = row1.backgroundSelectedColor;
```

Comment:
<!--
  0 failure: 
  1 warning:  tests/Resources/t...
  
  
  DangerID: danger-id-Danger;
  File: tests/Resources/ti.ui.tableview.test.js;
  Line: 1344;
-->

- :warning: tests/Resources/ti.ui.tableview.test.js line 1344 – 'value' is assigned a value but never used. (no-unused-vars)